File: 2DAM/ProgramacionDeServiciosYProcesos/Unidad-3/PrimeraConexionClienteServidor/src/main/java/es/drodriguez/com/Servidor.java
package es.drodriguez.com;

import java.io.DataOutput;
import java.io.DataOutputStream;
import java.io.IOException;
import java.net.ServerSocket;
import java.net.Socket;

public class Servidor {
    public static void main(String[] args) throws IOException {
        System.out.println("SERVIDOR");
        ServerSocket servidor;
        Socket socket;
        int contadorCliente = 0;
        int PUERTO = 5000;
        System.out.println("Esperando clientes...");

        try {
            servidor = new ServerSocket(PUERTO);
            do {
                contadorCliente++;
                socket = servidor.accept();
                System.out.println("Cliente conectado: " + contadorCliente);
                DataOutputStream salida = new DataOutputStream(socket.getOutputStream());
                salida.writeUTF("Hola cliente " + contadorCliente);
                salida.close();
                System.out.println("Se ha desconectado el cliente " + contadorCliente);

            }while (true);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: 2DAM/ProgramacionDeServiciosYProcesos/Unidad-3/CalculadoraSocketKotlin/src/main/kotlin/Servidor.kt
import java.net.ServerSocket
import java.net.Socket

fun main() {
    println("####### SERVIDOR ######")
    val PUERTO: Int = 5656
    val serverSocket: ServerSocket?
    var cliente: Socket?
    var numeroCliente: Int = 0

    try {
        serverSocket = ServerSocket(PUERTO)
        do {
            cliente = serverSocket.accept()
            println("Cliente: $cliente")
            numeroCliente++
            var hilo = Hilo(numeroCliente.toInt(), cliente)
            hilo.start()
        } while (true)


    } catch (e: Exception) {
        println(e.message)
    }
}